engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -227,4 +228,25 @@ public void refresh() {
         generateFrom(nodeList);
     }
 
+    /**
+     * Refreshes the task list if a refresh has been requested.
+     */
+    public void refreshIfNeeded() {
+        if (refreshRequested) {
+            refresh();
+
+            // Clear refresh requests to prevent refreshing unless needed
+            refreshRequested = false;
+        }
+    }
```

Comment:
That was actually my initial guess on how to do this, but won't the task list generator require a reference to the `WorldRendererImpl` now? I tried looking for existing ways that nodes could obtain a reference to the `WorldRenderer`, but couldn't find any.